xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/ResetPasswordManager.java

```diff
@@ -77,4 +78,14 @@ ResetPasswordRequestResponse checkVerificationCode(UserReference userReference,
      */
     void resetPassword(UserReference userReference, String newPassword)
         throws ResetPasswordException;
+
+    /**
+     * Ensure that the password matches the requirements provided by {@link RegistrationConfiguration}.
+     *
+     * @param newPassword the password to check
+     * @return {@code true} if the rules exposed in the {@link RegistrationConfiguration} are all respected.
+     * @since 15.10RC1
+     */
+    @Unstable
+    boolean isPasswordCompliantWithRegistrationRules(String newPassword);
```

Comment:
I'm surprised to see no corresponding revapi breakage in the pom.
Also, returning `true` by default could avoid to break revapi.

Reply:
Indeed, I thought about it when I implemented it and then I forgot and I guess we have a legacy module that I haven't built. I'll double check.